OpenRA.Mods.Common/Projectiles/Missile.cs

```diff
@@ -251,10 +255,23 @@ public Missile(MissileInfo info, ProjectileArgs args)
 
 			if (info.ContrailLength > 0)
 			{
+				hasContrail = true;
+
+				// Approximate maximum contrail length in WDist, used to calculate ScreenMap bounds.
+				// The *3 /2 magic is a HACK to work around the fact that speed * ContrailLength does not actually result in wide enough bounds.
+				// TODO: Figure out a way to calculate this more accurately and less hacky.
+				maxVisualContrailLength = new WDist((info.ContrailLength + info.ContrailDelay) * maxSpeed * 3 / 2);
```

Comment:
I might look into this, if you haven't tried solving it yet

Reply:
Feel free to, I probably won't have time before Sunday and have no idea how to tackle this anyway.